# Project 2

## Data Set 1 - Average Hourly Earnings

For this data set, the goal is to calculate the percent change in average hourly wages from year to year, as well as from month to month.

In [1]:
# Import pandas

import pandas as pan
from pandas import DataFrame, Series

# Read the information from .csv file into pandas

wages = pan.read_csv('BLS_Average_Hourly_Earnings.csv')

In [2]:
# Display average hourly earnings data in the dataframe

wages.head()

Year    Jan    Feb    Mar    Apr    May    Jun    Jul    Aug    Sep    Oct  \
0  2013  23.75  23.78  23.80  23.87  23.89  23.96  23.98  24.02  24.06  24.09   
1  2014  24.23  24.33  24.32  24.34  24.40  24.46  24.48  24.54  24.55  24.58   
2  2015  24.75  24.79  24.86  24.89  24.97  24.97  25.00  25.09  25.11  25.19   
3  2016  25.38  25.38  25.45  25.53  25.58  25.62  25.69  25.72  25.77  25.89   
4  2017  26.00  26.06  26.11  26.17  26.21  26.26  26.36  26.38  26.49  26.49   

     Nov    Dec  
0  24.16  24.18  
1  24.66  24.62  
2  25.25  25.24  
3  25.90  25.92  
4  26.52  26.61

The data is presented in a wide format, which does not indicate what the values between the two axes represent. To make this data easier to analyzed, the best course is to transform it into the long format using the .melt() function.

In [3]:
# Unpivot the wide format data to long format using the melt() function and reset the index

tidy_wage= wages.melt(id_vars=['Year'], var_name='Month', value_name = 'Hourly_Wage').sort_values(by=['Year']).reset_index(drop=True)
tidy_wage

Year Month  Hourly_Wage
0    2013   Jan        23.75
1    2013   Jun        23.96
2    2013   Apr        23.87
3    2013   Mar        23.80
4    2013   Oct        24.09
..    ...   ...          ...
127  2023   Mar          NaN
128  2023   Feb          NaN
129  2023   Jan        33.03
130  2023   Nov          NaN
131  2023   Dec          NaN

[132 rows x 3 columns]

The data has been converted to the long format. However, as can be seen, the Month column is not being sorted in chronological order. Given that the months are represented as strings and can therefore not be sorted properly using just the sort function, it makes sense to use a Categorical to enable proper sorting. 

In [4]:
# Use Categorical to enable proper sorting of string Month column. 

months = ['Jan', 'Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec']
tidy_wage['Month'] = pan.Categorical(tidy_wage['Month'], categories=months, ordered=True)
tidy_wage_sorted = tidy_wage.sort_values(by=['Year','Month'])
tidy_wage_sorted

Year Month  Hourly_Wage
0    2013   Jan        23.75
6    2013   Feb        23.78
3    2013   Mar        23.80
2    2013   Apr        23.87
7    2013   May        23.89
..    ...   ...          ...
123  2023   Aug          NaN
122  2023   Sep          NaN
121  2023   Oct          NaN
130  2023   Nov          NaN
131  2023   Dec          NaN

[132 rows x 3 columns]

Once the data has been tidied up and properly sorted, the calculations for the percent change in average hourly wages from month to month and year to year can be done. 

In [5]:
# Calculate the percent change in average hourly wage from month to month 

tidy_wage_sorted["percent_change"] = tidy_wage_sorted['Hourly_Wage'].pct_change()*100
tidy_wage_sorted

Year Month  Hourly_Wage  percent_change
0    2013   Jan        23.75             NaN
6    2013   Feb        23.78        0.126316
3    2013   Mar        23.80        0.084104
2    2013   Apr        23.87        0.294118
7    2013   May        23.89        0.083787
..    ...   ...          ...             ...
123  2023   Aug          NaN        0.000000
122  2023   Sep          NaN        0.000000
121  2023   Oct          NaN        0.000000
130  2023   Nov          NaN        0.000000
131  2023   Dec          NaN        0.000000

[132 rows x 4 columns]

Based on the analysis, it's clear that there is minimal changes in the average hourly wages from month to month.

In [6]:
# Calculate the average hourly wage for each year

avg_wage_yearly = tidy_wage_sorted.groupby('Year')['Hourly_Wage'].mean()
avg_wage_yearly

Year
2013    23.961667
2014    24.459167
2015    25.009167
2016    25.652500
2017    26.305000
2018    27.103333
2019    27.993333
2020    29.364167
2021    30.610833
2022    32.245833
2023    33.030000
Name: Hourly_Wage, dtype: float64

In [7]:
# Calculate the percent change in average hourly wages from year to year

avg_wage_yearly.pct_change()*100

Year
2013         NaN
2014    2.076233
2015    2.248646
2016    2.572390
2017    2.543612
2018    3.034911
2019    3.283729
2020    4.896999
2021    4.245537
2022    5.341246
2023    2.431839
Name: Hourly_Wage, dtype: float64

In [8]:
# Calculate the average percent change in average hourly wages between 2013 and 2023.

(avg_wage_yearly.pct_change()*100).mean()

3.2675142026634445

The percent change in average hourly wages from year to year range from a low of 2.07% to a high of 5.34%. The percent change in average hourly wages in the 10 years between 2013 and 2023 averages out 3.27%